presto-main/src/main/java/com/facebook/presto/cost/TableScanStatsRule.java

```diff
@@ -61,7 +62,10 @@ public TableScanStatsRule(Metadata metadata, StatsNormalizer normalizer)
         Map<VariableReferenceExpression, VariableStatsEstimate> outputVariableStats = new HashMap<>();
 
         for (Map.Entry<VariableReferenceExpression, ColumnHandle> entry : node.getAssignments().entrySet()) {
-            Optional<ColumnStatistics> columnStatistics = Optional.ofNullable(tableStatistics.getColumnStatistics().get(entry.getValue()));
+            // TODO Adjust statistics by taking into account subfield pruning possibly by changing
+            //  getTableStatistics SPI to pass in ColumnHandles for the columns listed in TableScanNode
+            ColumnHandle columnHandle = entry.getValue().withRequiredSubfields(ImmutableList.of());
```

Comment:
This makes some implicit assumptions about the `getTableStatistics` implementation. It feels like the list of column handles has to be passed to the `getTableStatistics`, and the connector itself should assign statistics for each of the handles passed.

Reply:
That's right. Would you like me to make that change in this PR?

Reply:
Yeah, i think it would be reasonable.